crypto/crypto.hpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (C) 2018 Bluzelle
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License, version 3,
+// as published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+//
+
+#pragma once
+
+#include <crypto/crypto_base.hpp>
+#include <options/options_base.hpp>
+#include <proto/bluzelle.pb.h>
+#include <openssl/evp.h>
+
+namespace bzn
+{
+    class crypto : public crypto_base
+    {
+    public:
+
+        crypto(std::shared_ptr<bzn::options_base> options);
+
+        void start();
```

Comment:
What's the reason behind a start? Can it not be done in the constructor?
Typically with use start because of enable_shared_from_this issues.